### PR TITLE
[Do not merge] Show taxonomy-based navigation in more cases 

### PR DESCRIPTION
**Pull request by tijmenb**
This is the current decision tree for when to show the taxonomy-based navigation:

image

This PR implements the following new logic:

image

[Diagrams on the Wiki](https://gov-uk.atlassian.net/wiki/spaces/GFED/pages/187924634/When+we+show+what+on+the+content+pages)

## Examples

- Page tagged to mainstream browse: https://government-frontend-pr-510.herokuapp.com/career-skills-and-training
- Page not tagged to browse, but tagged to new taxonomy: https://government-frontend-pr-510.herokuapp.com/government/publications/financial-support-for-full-time-students-of-higher-education-in-2016-to-2017
- Page not tagged to new taxonomy: https://government-frontend-pr-510.herokuapp.com/government/publications/the-olympic-and-paralympic-legacy-inspired-by-2012-second-annual-report

Supersedes https://github.com/alphagov/government-frontend/pull/506

https://trello.com/c/ZQNGFeUa

### CODE REVIEW

File: app/models/navigation_type.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
+class NavigationType
+  NEW_NAVIGATION_CONTENT_ITEM_SCHEMAS =
+    %w{answer contact guide detailed_guide document_collection publication}.freeze
+
+  def initialize(content_item)
+    @content_item = content_item
+  end
+
+  def should_present_new_navigation_view?
```
Comment (fofr) [inline app/models/navigation_type.rb]: `should_present_taxonomy_navigation`?

File: app/models/navigation_type.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+  def content_is_tagged_to_a_taxon?
+    @content_item.dig("links", "taxons").present?
+  end
+
+  def content_is_tagged_to_browse_pages?
+    @content_item.dig("links", "mainstream_browse_pages").present?
+  end
+
+  def content_schema_has_new_navigation?
```
Comment (fofr) [inline app/models/navigation_type.rb]: As mentioned in person, we should probably move away from `new_navigation`. Our "new navigation" will become our tested variants.

**Comment by tijmenb** [comment]
@fofr renamed everything in a separate commit.

**Comment by vanitabarrett** [comment]
This changes means there are less related links on the side compared to the B-variant that was running in the AB test.

**AB Test (B Variant):**
screen shot 2017-11-01 at 13 48 43


**This change:**
screen shot 2017-11-01 at 13 48 37

File: app/views/shared/_breadcrumbs.html.erb [hunk 3]
```diff
@@ -1,5 +1,3 @@
-<% # THIS IS PART OF EDUCATION NAVIGATION A/B TESTING - PLEASE KEEP IN SYNC WITH +new_navigation.erb %>
```
Comment (fofr) [inline app/views/shared/_breadcrumbs.html.erb]: Do we still need a warning about keeping templates in sync?
Seems like this logic could be moved into a presenter? Maybe in a future PR.

Comment (tijmenb) [reply app/views/shared/_breadcrumbs.html.erb]: I think people who'll modify those templates will be aware of the different variants, so I don't think we need the warning.

**Pull request merged**